### PR TITLE
fix: Add missing indicator columns to forms

### PR DESCRIPTION
**Pull request by vickiszhang**
Added 2 new fields 
- Reason for null
- Value Displayed

New fields added for forms, upload and export
Some fields are re-ordered
Subcategory changed to Topic
Tests updated

### CODE REVIEW

File: server/cpho/jinja2/indicators/upload_indicator.jinja2 [hunk 1]
```diff
@@ -146,17 +150,61 @@
...
+              <tr>
+                <td>{{ tdt("Data_Quality") }}</td>
+                <td>{{ tdt("Data Quality") }}</td>
+                <td>
+                  <ul>
+                    <li>{{ tdt("Caution") }}</li>
+                    <li>{{ tdt("Acceptable") }}</li>
+                    <li>{{ tdt("Good") }}</li>
+                    <li>{{ tdt("Suppressed") }}</li>
+                    <li>{{ tdt("Very Good")}}</li>
```
Comment (msarar) [inline server/cpho/jinja2/indicators/upload_indicator.jinja2:207]: load these in from the upload mapper
use the living arrangement PR for reference
https://github.com/PHACDataHub/cpho-phase2/pull/113/files#diff-c484d9604ef50ed4ea47591ff3d938b58d9ba2d25d32396bdb0f49e647f714ba

File: server/cpho/models/indicators.py [hunk 2]
```diff
@@ -177,20 +177,6 @@ class Meta:
...
-        on_delete=models.RESTRICT,
-    )
-
-    dimension_value = fields.ForeignKey(
-        "cpho.DimensionValue",
-        null=True,
-        blank=True,
-        on_delete=models.RESTRICT,
-    )
-
```
Comment (msarar) [inline server/cpho/models/indicators.py:193]: just re-ordering right?

File: server/cpho/views/upload_indicators.py
Comment (msarar) [inline server/cpho/views/upload_indicators.py:1]: i see the new fields in the upload help modal
but the upload does not actually consume the new columns
might need to add the new fields into import and export

File: server/cpho/views/view_util.py [hunk 3]
```diff
@@ -231,6 +231,76 @@ def export_mapper():
     return export_mapping
 
 
+def export_mapper():
```
Comment (msarar) [inline server/cpho/views/view_util.py:234]: something is wrong here:
There's two export mappers()

probably sth went wrong while resolving a merge conf

**Review by msarar — CHANGES_REQUESTED** [review]
Additional comment:
I appreciate renaming/reordering fields and their values
but if its a big change that touches multiple files, its best to do it in a separate branch with its own PR. Otherwise it gets quite confusing while reviewing the PR.
(note for the future)

Also, always a good idea to add a description to the PR
Ideally, list all the things the PR is improving/targeting

Another quick tip: always try to merge in main periodically instead of once at the end. Its easier to resolve many small merge confs than one large merge conf

File: server/cpho/models/indicators.py (hunk 2 above)
Comment (andGuo) [reply server/cpho/models/indicators.py:193]: Yup 👍

File: server/cpho/jinja2/indicators/upload_indicator.jinja2 [hunk 4]
```diff
@@ -6,7 +6,11 @@
   <button type="button mt-5"
           class="btn btn-primary"
           data-bs-toggle="modal"
-          data-bs-target="#expectedColumnModal">{{ tdt("View expected column names") }}</button>
+          data-bs-target="#expectedColumnModal">
+    {{ tdt("View expected column names") }}
+  </button>
+ 
+  <a class="btn btn-primary" href="{{ url("export_indicator_template") }}">{{ tdt("Download a file template") }}</a>
```
Comment (msarar) [inline server/cpho/jinja2/indicators/upload_indicator.jinja2:13]: duplicate

File: server/cpho/jinja2/indicators/upload_indicator.jinja2 [hunk 5]
```diff
@@ -145,17 +143,32 @@
...
+                    {% for key in mapper["value_unit_mapper"] %}
+                      {% if key != "" %}<li>{{ tdt(key) }}</li>{% endif %}
+                    {% endfor %}
+                  </ul>
+                </td>
+              </tr>
+              <tr>
+                <td>{{ tdt("Value_Displayed") }}</td>
+                <td>{{ tdt("Value Displayed") }}</td>
+                {% for key in mapper["value_displayed_mapper"] %}
```
Comment (msarar) [inline server/cpho/jinja2/indicators/upload_indicator.jinja2:159]: Should be inside a <td><ul></ul></td>

File: server/cpho/jinja2/indicator_data/manage_indicator_data.jinja2 [hunk 6]
```diff
@@ -78,12 +81,15 @@
...
               <th width="9%">{{ tdt("Value Upper Bound") }}</th>
-              <th width="10%">{{ tdt("Data Quality") }}</th>
-              <th width="12%">{{ tdt("Single Year Timeframe") }}</th>
-              <th width="12%">{{ tdt("Multi Year Timeframe") }}</th>
+              <th width="9%">{{ tdt("Data Quality") }}</th>
+              <th width="9%">{{ tdt("Single Year Timeframe") }}</th>
+              <th width="9%">{{ tdt("Multi Year Timeframe") }}</th>
+              <th width="9%">{{ tdt("Value Displayed") }}</th>
+              <th width="9%">{{ tdt("Pt Data Availability") }}</th>
+              <th width="9%">{{ tdt("Age Group Type") }}</th>
```
Comment (msarar) [inline server/cpho/jinja2/indicator_data/manage_indicator_data.jinja2:92]: remove this line: dont need age group type in predefined_dimensions formset

File: server/cpho/jinja2/indicator_data/manage_indicator_data.jinja2 [hunk 7]
```diff
@@ -133,6 +139,9 @@
                     </div>
                   {% endif %}
                 </td>
+                <td>{{ form.value_displayed }}</td>
+                <td>{{ form.pt_data_availability }}</td>
+                <td>{{ form.age_group_type }}</td>
```
Comment (msarar) [inline server/cpho/jinja2/indicator_data/manage_indicator_data.jinja2:144]: remove this line: dont need age group type in predefined_dimensions formset

**Review by msarar — APPROVED** [review]

File: server/cpho/schema.py [hunk 8]
```diff
@@ -100,7 +100,7 @@ def resolve_indicator(root, info, **kwargs):
 class CreateIndicator(graphene.Mutation):
     class Arguments:
         category = graphene.String(required=True)
-        sub_category = graphene.String(required=True)
```
Comment (AlexCLeduc) [inline server/cpho/schema.py:103]: no need to touch the graphql stuff right now, it's all being re-written

**Review by AlexCLeduc — APPROVED** [review]

File: server/cpho/schema.py (hunk 8 above)
Comment (msarar) [reply server/cpho/schema.py:103]: GraphQL stuff is still pretty much the same except for the subcategory field that got changed to topic.
ShouldI checkout schema.py from main to revert the changes? or keep it as it is?

Comment (AlexCLeduc) [reply server/cpho/schema.py:103]: I guess it doesn't matter, it'll be a full rewrite anyway

**Pull request merged**